rust/gui-client/docs/process_split.md

```diff
@@ -0,0 +1,39 @@
+# Process split
+
+This is meant for Linux, but it will probably be very similar on Windows.
+
+## TODO
+
+- Do we need to support CLI and GUI Clients being installed on the same system? The tunnel binary would conflict if they both try to install it. Splitting it into 3 packages would complicate downloads.
+- Clients and Gateways can't run in the same system, right? How would DNS work if they did?
+- If there is a service account token, but it's invalid, should the tunnel wait for the GUI to connect, or fail? Will the tunnel know whether a GUI is even installed?
+
+## Binaries
+
+I gave them new names to clarify.
+
+1. `firezoned` - A daemon that runs the tunnel and has elevated privilege
+2. `firezonectl` - A CLI program that runs unprivileged and exits once each operation is complete, like `systemctl` or `nmcli`
+3. `firezone-gui` - An unprivileged GUI program that keeps an icon in the system tray / notification center and is similar to `firezonectl`
+
+## Docker
+
+`firezoned` runs all the time. You can do `docker exec` to run `firezonectl` within the container.
```

Comment:
Need to think more about the use case behind running a client in a container. I think it would only make sense with `--network host` so that the routes it adds affect the whole system.

Reply:
Yeah. Then I guess the appeal is using docker-compose as an alternative to systemd. Both provide auto-start and sandboxing, but maybe some users happen to want everything in Docker for uniformity. Plus Docker tending to be opt-in for security vs systemd tending to be opt-out. (A secure service unit file is long)

Reply:
agreed